### ISSUE TITLE
Slow operations over NAPI

### ISSUE THREAD

**Issue by ncannasse** [microsoft/vscode#138467]
Hi !

I'm the main developer of the Hashlink Debugger (https://github.com/vshaxe/hashlink-debugger)
We have noticed after upgrading from VSCode 1.55 to 1.56 that there was an important slowdown of the debugger.

I now took time to investigate it further, and found that the issue was related to a single call being made through ffi-napi to read the process memory (on windows ReadProcessMemory - https://github.com/vshaxe/hashlink-debugger/blob/master/hld/NodeDebugApi.hx#L46). Every call now takes tens to hundreds of milliseconds to complete which was not the case in VSCode 1.55

I thought first it was a performance regression in libffi so I upgraded to use another ffi library which claims superior speed (sbffi : https://github.com/bengl/sbffi), but the results are again good on 1.55 but very slow on latest 1.62.3, which leads me to think there is something in either VSCode or Electron that has been changed regarding NAPI implementation which makes ffi usage a magnitude slower than it used to be, making the debugger not very usable anymore. 

Any insight to solve this quite technical issue would be greatly appreciated.

**Comment by gjsjohnmurray** [comment]
Looks like 1.56 was when Electron 12 landed.

https://code.visualstudio.com/updates/v1_56#_electron-12-update

**Comment by deepak1556** [comment]
This would be a side-effect of https://github.com/electron/electron/pull/28957 that requires https://github.com/nodejs/node/commit/7abc7e45b2e177307f67f33906663b5de02f2916. Can you check if the issue is present with exploration builds based on newer runtime https://github.com/microsoft/vscode/pull/137241 that contains the above fixes.

[macOS universal](https://az764295.vo.msecnd.net/exploration/1ed4b16d638968489596d86fc737e05dd10e052d/VSCode-darwin-universal.zip)
[windows x64 user setup](https://az764295.vo.msecnd.net/exploration/1ed4b16d638968489596d86fc737e05dd10e052d/VSCodeUserSetup-x64-1.63.0-exploration.exe)

**Comment by ncannasse** [comment]
I tried the exploration build but it still has the same behavior.

This can be reproduce by installing Haxe extension as well as Hashlink debugger, then creating a new project with these files :
[testproject.zip](https://github.com/microsoft/vscode/files/7663061/test.zip)

Put a breakpoint on Main.hx line 37 and compile & run. When the debugger breaks, try to display the content of `o.m` (a map with 200 elements). Notice a long loading time due to calls to ReadProcessMemory. Reproduced on Windows 10.
